conflation/trace_fetching/mapillary.py

```diff
@@ -198,181 +206,332 @@ def make_trace_data_requests(
+                    sequence_id = values[feature.tags[i + 1]].string_value
+            if captured_at and captured_at > start_date_epoch:
+                if sequence_id and sequence_id not in seen_sequences:
+                    seen_sequences.add(sequence_id)
+                    sequence_resp = session_.get(
+                        SEQUENCE_URL.format(sequence_id, access_token)
                     )
 
-                # Check if there is a next image page or if we are finished with this sequence
-                img_next_url = (
-                    img_resp.links["next"]["url"] if "next" in img_resp.links else None
-                )
-                img_page += 1
+                    image_ids = [
+                        img_id_obj["id"] for img_id_obj in sequence_resp.json()["data"]
+                    ]
 
-        # Already collected enough sequences. Move onto the next bbox section
-        if len(sequences_by_id) > max_sequences_per_bbox_section:
-            print(
-                "## Already collected {} seqs for this bbox section, greater than max_sequences_per_bbox_section={}. "
-                "Continuing...".format(len(sequences_by_id), max_sequences_per_bbox_section)
-            )
-            break
+                    # Skip sequences that have too few images
+                    if len(image_ids) < skip_if_fewer_imgs_than:
+                        continue
 
-        # Check if there is a next sequence page or if we are finished with this bbox
-        seq_next_url = seq_resp.links["next"]["url"] if "next" in seq_resp.links else None
-        seq_page += 1
+                    images_resp = session_.get(
+                        IMAGES_URL.format(",".join(image_ids), access_token)
+                    )
+                    images = [
+                        {  # Convert to seconds because filtering / map matching assumes time in seconds
+                            "time": img_obj["captured_at"] / 1000,
+                            "lon": img_obj["geometry"]["coordinates"][0],
+                            "lat": img_obj["geometry"]["coordinates"][1],
+                        }
+                        for img_obj in images_resp.json()["data"]
+                    ]
 
-    print("Keys: {}".format(list(sequences_by_id.keys())))
+                    # Mapillary returns their trace data in random chronological order, so we need to sort the images
+                    images = sorted(images, key=lambda x: x["time"])
 
-    # We don't care about the sequence IDs anymore (just using it as a method to group trace data), so we just return
-    # values
-    sequences = list(sequences_by_id.values())
+                    # Only process sequences that originated from this bbox. This prevents us from processing sequences
+                    # twice
+                    origin_lon, origin_lat = (
+                        images[0]["lon"],
+                        images[0]["lat"],
+                    )
```

Comment:
We sort the images chronologically and use the origin lon, lat to check if it's within this zoom 14 tile to remove duplicates.